src/parameterizations/vertical/MOM_opacity.F90

```diff
@@ -353,13 +377,44 @@ subroutine opacity_from_chl(optics, sw_total, sw_vis_dir, sw_vis_dif, sw_nir_dir
         do n=1,nbands
           optics%sw_pen_band(n,i,j) = Inv_nbands*sw_pen_tot
         enddo
-      enddo ; enddo
+     enddo; enddo
+    case (OHLMANN_03)
+      ! want exactly two penetrating bands. If not, throw an error.
+      if ( nbands /= 2 ) then
+        call MOM_error(FATAL, "opacity_from_chl: CS%opacity_scheme requires nbands==2.")
+      endif
+      !$OMP parallel do default(shared) private(SW_vis_tot)
+      do j=js,je ; do i=is,ie
+        SW_vis_tot = 0.0  ! Ohlmann does not classify as vis/nir. Using vis to add up total
+        if (G%mask2dT(i,j) < 0.5) then
+          optics%sw_pen_band(1:2,i,j) = 0.  ! Make sure there is a valid value for land points
+        else
+          if (multiband_vis_input ) then ! If multiband_vis_input is true then so is multiband_nir_input
+             SW_vis_tot = sw_vis_dir(i,j) + sw_vis_dif(i,j) + &
```

Comment:
Please add parentheses to specify the order of the sum of 4 real variables here so that the answers are not unnecessarily sensitive to the level of optimization.  This really helps with debugging.

  Also, please follow the 2-space indenting convention used throughout the MOM6 code, as described in the MOM6 style guide at https://github.com/mom-ocean/MOM6/wiki/Code-style-guide#white-space, both here and init_ohlmann_table later in this file.